service/common/src/main/java/org/apache/polaris/service/auth/DefaultOAuth2ApiService.java

```diff
@@ -75,43 +74,39 @@ public Response getToken(
     if (!tokenBroker.supportsRequestedTokenType(requestedTokenType)) {
       return OAuthUtils.getResponseFromError(OAuthTokenErrorResponse.Error.invalid_request);
     }
-    if (authHeader == null && clientId == null) {
+    if (authHeader == null && clientSecret == null) {
       return OAuthUtils.getResponseFromError(OAuthTokenErrorResponse.Error.invalid_client);
     }
-    if (authHeader != null && clientId == null && authHeader.startsWith("Basic ")) {
+    // token exchange with client id and client secret means the client has previously
+    // attempted to refresh an access token, but refreshing was not supported by the token broker.
+    // Accept the client id and secret and treat it as a new token request
+    if (authHeader != null && clientSecret == null && authHeader.startsWith("Basic ")) {
```

Comment:
Sorry, but I'm not sure how this logic aligns with the comment above. "With ... client secret", but the secret is `null` here?... Would you mind clarifying the comment?

Reply:
I'm fine with this change as it is obviously meant to support existing Iceberg REST client, but I do not think this logic follows the client credentials flow in RFC [6749](https://www.rfc-editor.org/rfc/rfc6749#page-40). Specifically, "client secret" is not supposed to be passed in the POST body (from where the initial value for `clientSecret` comes, if I'm not mistaken).

Reply:
To be complete, [RFC 6749 section 2.3.1]( https://www.rfc-editor.org/rfc/rfc6749#section-2.3.1) permits client secret in the request body, but does not recommend it.

Reply:
Thanks, @adutra . I missed that option :facepalm:

Reply:
In any case, I think Polaris ought to treat client ID and secret as a tuple and not mix ID from header and secret from POST body (or the other way around)... which seems possible ATM.

Reply:
> Sorry, but I'm not sure how this logic aligns with the comment above. "With ... client secret", but the secret is `null` here?... Would you mind clarifying the comment?

I actually just moved the comment from below. But I'll add that the clientId and secret come from the auth header in this case.
https://github.com/apache/polaris/pull/952/files/4cd4130b286d327428bc660e3731234561ab0d1e#diff-3e1d745f65cde4f3699177beac21b387a0346daef83978ec835c7037ddb9b8baL100